httpcore/_async/connection.py

```diff
@@ -30,6 +34,7 @@ def __init__(
         self.connect_failed = False
         self.expires_at: Optional[float] = None
         self.backend = AutoBackend()
+        self.socket = socket
```

Comment:
Nit: should we move this next to the other attributes that directly come from parameters (i.e. below `.ssl_context`)?